symbolic-base/src/ZkFold/Symbolic/Data/FFA.hs

```diff
@@ -196,3 +208,5 @@ instance (Prime p, Symbolic c) => Field (FFA p c) where
 
 instance Finite (Zp p) => Finite (FFA p b) where
   type Order (FFA p b) = p
+
+deriving newtype instance Symbolic c => Eq (Bool c) (FFA p c)
```

Comment:
The problem with current implementation of FFA, among many (performance, inefficient implementation of inverses,...), is that it is imprecise: the value is encoded up to the `+ p k` for some `k`.
So this `Eq` instance is incorrect, to make it work, we need to basically rewrite `FFA` from scratch to the more efficient implementation which is now possible thanks to the new `MonadCircuit` interface: https://hackmd.io/@kullervo/cryptographyExplorations/%2F%40kullervo%2FbignumberCircuits